fix: wallets with a lot of tokens were making the bx to crash

estebanmino opened the pull request:
Fixes BX-####
Figma link (if any):

## What changed (plus any additional context for devs)

@magiziz found an issue where wallets with many assets as vitalik.eth were crashing the bx
@magiziz also provided repro steps, during the tanstack virtualized list package bump I didn't set the height of the tokens tab, so the ui was rendering the whole list of tokens all the time, causing the bs to be very slow and crash

this pr now sets a height of the tokens tab so the virtualized list works as expected
i didn't make this change to activity nor nfts since those have pagination and work differently

thanks @magiziz  🐐 

## Screen recordings / screenshots

<!-- Screen recordings can also be helpful for showing reviewers what to test for.  -->

## What to test

<!--

Please be thorough about what to test to help reviewers.
You might want to emphasize potential regressions to check for.
If your code relies on a feature flag for checking both paths of the feature flag, other parts of the code that may have been impacted by your changes, etc.

Don't know what to write here? List all the steps you did to test the changes. This might help QA better understand what/how to test.

-->

try importing vitalik.eth and check everything works fine again

DanielSinclair commented on static/data/networks.json:
@estebanmino Were these json test files included by accident?

estebanmino replied:
uh yes

DanielSinclair reviewed (changes requested):
Found a small padding issue at the top of the list:
<img width="353" alt="Screenshot 2024-06-07 at 6 46 45 PM" src="https://github.com/rainbow-me/browser-extension/assets/4412473/ac4c40cf-423c-44a4-b2b6-d15767b20089">

There seems to be something off when reaching the bottom of the scroll list with cells going missing; was looking at domm.eth here. The header seems to not want to fully collapse too; maybe scroll hijacking issue
https://github.com/rainbow-me/browser-extension/assets/4412473/43b4727c-1cdd-43c2-8863-6dee4e0293d8

DanielSinclair commented:
Super speedy ⚡

magiziz commented:
One thing @estebanmino. Just noticed this.

The assetsRowVirtualizer.getVirtualItems() in <Tokens />

magiziz commented:
One thing @estebanmino. Just noticed this.

The assetsRowVirtualizer.getVirtualItems() in <Tokens /> for some reason doesn't get virtualized. It should show like 13 items and then once user scrolls it should show more, but it shows 2.9k for some reason.

estebanmino commented:
taking a look to both comments rn

estebanmino commented:
@DanielSinclair fixed the padding issue you mentioned, using virtualizer padding instead of wrapper component padding
@magiziz i'm having issues reproducing what you see, virtualized lists are showing up as expected


https://github.com/rainbow-me/browser-extension/assets/12115171/00fa5596-6e23-41c1-a494-f197cace9469

magiziz commented:
@estebanmino Ah my bad. I probably didn't pull the latest changes, looks good 👍

DanielSinclair commented:
@estebanmino Still seeing some issues on this mainly with invisible scroll and the header getting stuck while collapsing:


https://github.com/rainbow-me/browser-extension/assets/4412473/eab37b03-372c-4fc1-ae17-d57cbc8f842f

estebanmino commented:
@DanielSinclair what a weird bug, i moved ids and the way we use height and i can't reproduce it anymore, could you try again to see if you still see it?


https://github.com/rainbow-me/browser-extension/assets/12115171/6a394fd5-2d9f-4bac-b0c1-8cabe82d157f

BrodyHughes commented:
checked out a couple big wallets and looks very good. the speed is so much better. only a couple seconds to load vitalik.eth is crazy 🤯

BrodyHughes commented on src/entries/popup/pages/home/Tokens.tsx:
```diff
@@ -223,11 +223,14 @@ export function Tokens() {
 
   const containerRef = useContainerRef();
+
+  console.log('filteredAssets', filteredAssets.length);
```
👀

DanielSinclair commented on src/entries/popup/pages/home/Tokens.tsx:
```diff
@@ -259,6 +262,7 @@ export function Tokens() {
     <Box
       width="full"
       style={{
+        height: `336px`,
```
Seeing a small padding issue at the top of the list:
<img width="190" alt="Screenshot 2024-06-19 at 2 44 01 PM" src="https://github.com/rainbow-me/browser-extension/assets/4412473/36d20f6c-8759-44b7-bf0f-7e1bbf8b8731">

DanielSinclair commented:
Looks like there is still an issue with the header collapse on scroll. Occasionally I'll see the token list disappear as well. The easiest way to reproduce that seems to be to refresh the extension in Chrome's extension manager, login, then attempt to scroll. You can get out of this state and see the list properly with the `r` refresh hotkey

https://github.com/rainbow-me/browser-extension/assets/4412473/364f598c-66c7-493b-bf78-ce6103596329

<img width="360" alt="Screenshot 2024-06-19 at 2 46 30 PM" src="https://github.com/rainbow-me/browser-extension/assets/4412473/dd7f2bba-e347-4914-b686-3b7b394cf1b9">

DanielSinclair reviewed (changes requested):
Empty list issue seems to be resolved! I occasionally see a flash of the cells on scroll, but the only blocker I see is that the scroll header won't collapse. If you hover over the header component, it will scrolljack on that component and move ~10 pixels
https://github.com/rainbow-me/browser-extension/assets/4412473/6181712e-0e2e-44d9-a828-a428565bed88

DanielSinclair commented:
@estebanmino Much better! I think we just need to make changes to Tokens to replicate the Activity behavior (collapsing header), and then a small padding issue at the bottom of NFTs 

https://github.com/rainbow-me/browser-extension/assets/4412473/02429237-7291-4d0a-80c8-e1b7e2d92fa6

<img width="363" alt="Screenshot 2024-07-08 at 7 35 50 PM" src="https://github.com/rainbow-me/browser-extension/assets/4412473/8a6bf2c3-df34-4167-951e-e14df75bfd04">

estebanmino commented:
@DanielSinclair couldn't open your video but i think i fixed what you were saying


https://github.com/rainbow-me/browser-extension/assets/12115171/793980b3-f4a4-44e3-a3ba-e37a6ed7d336


i also found that `useContainerRef` was making `useVirtualizer` to not show the whole list sometimes, so i changed it to `useRef`
